### ISSUE TITLE
[Bug] Site works properly in normal tab, breaks in private tab

### ISSUE THREAD

**Issue by markochk** [mozilla-mobile/fenix#16770]
[This site](https://m.kupujemprodajem.com/index.php) works properly in normal tab, but has odd behavior in private tab. I've uploaded a [video](https://gfycat.com/harmfulcorruptleech) where I first open this site in normal tab, where searching works fine, and then opening the same site in private tab, where searching breaks it. Afterwards, Firefox is responsive and can open new tabs just fine.

## Steps to reproduce

1. open https://m.kupujemprodajem.com/index.php in private tab
2. after it's done loading, try either searching in it and pressing enter, refreshing the page, or searching in address bar
3. page goes black and tab disappears from private tab list

There's no difference with `Enhanced Tracking Protection` turned off.

### Expected behavior

Page should work properly, like it does when opened in non-private tab.

### Actual behavior

Page isn't responsive and disappears from private tab list.

### Device information

* Huawei Honor 6X (Android 9)
* Firefox Nightly 201128 17:02 (Build #2015778315) 
* Add-ons installed: uBlock Origin, Decentraleyes

**Comment by LaurentiuApahideanSV** [comment]
I can confirm the issue is reproducible Firefox Preview Nighlty 201207.
### Devices used
- OnePlus6T (Android 9)
- Samsung Galaxy S9 (Android 8.0.0)

**Comment by markochk** [comment]
I should've given an update a month ago, but I forgot, so apologies for that.

About a month ago I noticed that this site started working properly in private mode and has been working properly since. I've tested every action mentioned in step 2 and haven't been able to reproduce the crash (step 3).

Site does have an issue where the footer overlaps the first 3 results, but that is unrelated to this issue and happens in both normal and private Firefox mode and is easily fixed with uBlock Origin's element picker.

Running Firefox Nightly 94.0a1 (Build #2015833611)

Thank you!
